webv2/table/remove_column.go

```diff
@@ -0,0 +1,188 @@
+
+	sp = removeColumnFromSpannerSecondaryIndex(sp, colName)
+
+	sp = removeColumnFromSpannerForeignkeyColumns(sp, colName)
+
+	sp = removeColumnFromSpannerForeignkeyReferColumns(sp, colName)
+
+	sp = removeColumnFromSpannerColNames(sp, colName)
+
+	removeSpannerSchemaIssue(table, colName, Conv)
+
+	removeColumnFromToSpannerToSource(table, colName, Conv)
+
+	Conv.SpSchema[table] = sp
+}
+
+// removeColumnFromSpannerColNames remove given column from ColNames.
+func removeColumnFromSpannerColNames(sp ddl.CreateTable, colName string) ddl.CreateTable {
+
+	for i, col := range sp.ColNames {
+		if col == colName {
+			sp.ColNames = utilities.Remove(sp.ColNames, i)
+			break
+		}
+	}
+	delete(sp.ColDefs, colName)
+	return sp
+}
+
+// removeColumnFromSpannerPK remove given column from Primary Key List.
+func removeColumnFromSpannerPK(sp ddl.CreateTable, colName string) ddl.CreateTable {
+
+	for i, pk := range sp.Pks {
+		if pk.Col == colName {
+			sp.Pks = utilities.RemovePk(sp.Pks, i)
+			break
+		}
+	}
+	return sp
+}
+
+// removeColumnFromSpannerColDefs remove given column from Spanner ColDefs List.
+func removeColumnFromSpannerColDefs(sp ddl.CreateTable, colName string) ddl.CreateTable {
+	delete(sp.ColDefs, colName)
+	return sp
+}
+
+// removeColumnFromSpannerSecondaryIndex remove given column from Spanner SecondaryIndex List.
+func removeColumnFromSpannerSecondaryIndex(sp ddl.CreateTable, colName string) ddl.CreateTable {
+
+	for i, index := range sp.Indexes {
+		for j, key := range index.Keys {
+			if key.Col == colName {
+				sp.Indexes[i].Keys = removeColumnFromSecondaryIndexKey(sp.Indexes[i].Keys, j)
+				break
+			}
+		}
+	}
+	return sp
+}
+
+// removeColumnFromSecondaryIndexKey remove given column from SpannerSecondary Index Key List.
+func removeColumnFromSecondaryIndexKey(slice []ddl.IndexKey, s int) []ddl.IndexKey {
+	return append(slice[:s], slice[s+1:]...)
+}
+
+// removeColumnFromSecondaryIndexKey remove given column from Spanner Secondary Schema Issue List.
+func removeSpannerSchemaIssue(table string, colName string, Conv *internal.Conv) {
+	if Conv.Issues != nil {
+		if Conv.Issues[table] != nil && Conv.Issues[table][colName] != nil {
+			delete(Conv.Issues[table], colName)
+		}
+	}
+}
+
+// removeColumnFromToSpannerToSource remove given column from ToSpanner and ToSource List.
+func removeColumnFromToSpannerToSource(table string, colName string, Conv *internal.Conv) {
+
+	srcTableName := Conv.ToSource[table].Name
+
+	srcColName := Conv.ToSource[table].Cols[colName]
+	delete(Conv.ToSource[table].Cols, colName)
+	delete(Conv.ToSpanner[srcTableName].Cols, srcColName)
+}
+
+// removeColumnFromSpannerForeignkeyColumns remove given column from Spanner Foreignkey Columns List.
+func removeColumnFromSpannerForeignkeyColumns(sp ddl.CreateTable, colName string) ddl.CreateTable {
+
+	for i, fk := range sp.Fks {
+		for j, column := range fk.Columns {
+			if column == colName {
+				sp.Fks[i].Columns = removeFkColumns(fk.Columns, j)
```

Comment:
once a foreign key is removed the index's element will be modified and the subsequent element would be missed, we will have to handle that scenario

Reply:
@shreyakhajanchi Could you please elaborate a bit more on this.

Reply:
Updated. Please review it